pulp_ansible/app/models.py

```diff
@@ -178,6 +179,38 @@ class Meta:
         ]
 
 
+class CollectionVersionSignature(Content):
+    """
+    A content type representing a signature that is attached to a content unit
+
+    Fields:
+        data (models.BinaryField): A signature, base64 encoded. # Not sure if it is base64 encoded
+        digest (models.CharField): A signature sha256 digest.
+        pubkey_fingerprint (models.CharField): A fingerprint of the public key used.
+
+    Relations:
+        signed_collection (models.ForeignKey): A collection version this signature is relevant to.
+        signing_service (models.ForeignKey): An optional signing service used for creation.
+    """
+
+    PROTECTED_FROM_RECLAIM = False
+    TYPE = "collection_signature"
+
+    signed_collection = models.ForeignKey(
+        CollectionVersion, on_delete=models.CASCADE, related_name="signatures"
+    )
+    data = models.BinaryField()  # This is what the pulp_container folks used
+    digest = models.CharField(max_length=64)  # does adding a min length improve db efficiency?
+    pubkey_fingerprint = models.CharField(max_length=64)
```

Comment:
I kept thinking about this, and i think it is ok, if we can deliver the name of the signer (key user_id) for the keys we know and say unknown for keys we don't have/care about.
So this is kind of a voluntary foreign key into a possible future key database.

Reply:
How are you thinking of getting the key user id into the signature model? Would a new field have to be added to the SigningService, or maybe a new method to extract it from the public key (although I don't think key user id is a requirement for the public key, atleast in OpenPGP)?

Reply:
I would not put it in the signature model at all. We have the key in the signing service so we can either add a user id field there or extract from the key field on the go. We would report unknown for keys we don't have.

Reply:
I thought the value of this was in cases where the signature wasn't made by a Pulp SigningSerice, but instead sync'd from another Pulp where the signature was produced.

Reply:
> I thought the value of this was in cases where the signature wasn't made by a Pulp SigningSerice, but instead sync'd from another Pulp where the signature was produced.

That is why i think we need a pubkey model eventually.